pkg/providers/tinkerbell/validator.go

```diff
@@ -227,6 +228,15 @@ func (v *Validator) ValidateMinimumRequiredTinkerbellHardwareAvailable(spec v1al
 	return nil
 }
 
+// ValidateHardwaresAreFound ensures that there are one or more hardware CRDs presnt in the hardware list for a cluster.
+func (v *Validator) ValidateHardwaresAreFound(hardwares []tinkv1alpha1.Hardware) error {
```

Comment:
Why does a consumer need to call this vs `if len(hardwares.Hardware) != 0`?

Reply:
I initially just had that `if` inside `SetupAndValidateDeleteCluster`. But then thought to maybe follow the practice of having a separate validation method inside `validator`. 
I do agree its a bit much, I could change it back and later if we have more things to validate we can move it out 👍

Reply:
Another reason I thought about was for padding the `error` message. 
`failed setup and validations: no hardware found for cluster`